app/services/file_locator.rb

```diff
@@ -89,17 +90,29 @@ def location
     end
   end
 
-  # If S3, download object to /tmp
+  # If S3 or http(s), download object to /tmp
   def local_location
     @local_location ||= begin
-      if uri.scheme == 's3'
+      case uri.scheme
+      when 's3'
         S3File.new(uri).local_file.path
-      else
+      when 'file'
         location
+      else
+        local_file.path
       end
     end
   end
 
+  def local_file
+    @local_file ||= Tempfile.new(filename)
+    File.binwrite(@local_file, reader.read)
+    @local_file.rewind
```

Comment:
Is this rewind necessary given the ensure closes the file?

Reply:
Right, good catch. It does not look like it is necessary.